src/ai/backend/manager/types.py

```diff
@@ -41,3 +41,18 @@ class UserScope:
 
 class DistributedLockFactory(Protocol):
     def __call__(self, lock_id: LockID, lifetime_hint: float) -> AbstractDistributedLock: ...
+
+
+class RaftNodeInitialRole(str, enum.Enum):
+    LEADER = "leader"
+    VOTER = "voter"
+    LEARNER = "learner"
```

Comment:
If raftify or raft-rs define the behaviors for each role, how about having those roles in raftify?

Reply:
In fact, in Raftify, `InitialRole` type is already defined on the Rust side, and this type exists in the Python bindings as well. 

The reason I define and use a separate Enum with the same contents here is because the type is not an Enum in Python, which makes it unusable as is in the Python code.

For instance, below type validating code produce the following erorr.

```py
from raftify import InitialRole
...
	t.Key("role", default=InitialRole.VOTER): tx.Enum(InitialRole),
```

```
  File "/home/jopemachine/backend.ai/dist/export/python/virtualenvs/python-default/3.12.2/lib/python3.12/site-packages/trafaret/base.py", line 1143, in transform
    for k, v, names in key(value, context=context):
  File "/home/jopemachine/backend.ai/dist/export/python/virtualenvs/python-default/3.12.2/lib/python3.12/site-packages/trafaret/base.py", line 972, in __call__
    result = self.trafaret(self.get_data(data, default), context=context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jopemachine/backend.ai/dist/export/python/virtualenvs/python-default/3.12.2/lib/python3.12/site-packages/trafaret/base.py", line 152, in __call__
    return self.check(val, context=context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jopemachine/backend.ai/dist/export/python/virtualenvs/python-default/3.12.2/lib/python3.12/site-packages/trafaret/base.py", line 115, in check
    return self.check_and_return(value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jopemachine/backend.ai/src/ai/backend/common/validators.py", line 216, in check_and_return
    return self.enum_cls(value)
           ^^^^^^^^^^^^^^^^^^^^
TypeError: No constructor defined
```